docs/DeveloperGuide.md

```diff
@@ -1,34 +1,503 @@
+    * Locate the **build.gradle file** (not the root folder as you would do in a normal importing) and select it. Click `OK`.
+    * If asked, choose to `Open as Project` (not Open as File).
+    * Click `OK` to accept the default settings but do ensure that the selected version of Gradle JVM matches the 
+    JDK being used for the project.
+    * Wait for the importing process to finish (may take a few minutes, so get some :coffee: :relaxed:)
+1. **Verify the set up**:
+    * Run the `..main/java/seedu.duke/Duke`
+    * Upon a successful configuration, you should be able to see the welcome screen of the project as shown below:
+    ![verify_set_up](images/Verify_setup.PNG)
+    * Run some tests to ensure it is working properly.
+
+### Getting started
+
+1. **Configure the coding style**
+If using IDEA, follow the guide If using IDEA, follow the guide [se-edu/guides IDEA: Configuring the code style](https://se-education.org/guides/tutorials/intellijCodeStyle.html) 
+to set up IDEA’s coding style to match ours.
+
+    :bulb: Tips: Optionally, you can follow the guide  [se-edu/guides Using Checkstyle](https://se-education.org/guides/tutorials/checkstyle.html) 
+    to find how to use the CheckStyle within IDEA e.g., to report problems as you write code.
+1. **Set up CI**
+This project comes with a GitHub Actions config files (in `.github/workflows` folder). When GitHub detects those files, 
+it will run the CI for your project automatically at each push to the master branch or to any PR. No set up required.
+1. **Learn the design**
+When you are ready to start coding, we recommend that you get some sense of the overall design by reading about [*25HoursADay*’s architecture](#architecture).
+
+## Design
+This section describes the design overview of the *25HoursADay* application.
+
+### Architecture
+
+The figure below shows the overall design of the application. 
+<img src="https://raw.githubusercontent.com/AY2021S1-CS2113T-T12-2/tp/master/images/Overall_Architecture.JPG" alt="" width="300"/> <br/>
+
+The sequence diagram below shows how the components interact with each other for the scenario where the user issues the command `todo`.
+<img src="https://raw.githubusercontent.com/AY2021S1-CS2113T-T12-2/tp/master/images/Archi_SD.JPG" alt="" width="750"/>
+
+An overview of each components is listed below.
+
+### Ui
+The `Ui` class is responsible for the user interface of the application. It receives the input from the user and prints the results to the user.
+
+### Parser
+
+The `Parser` class is responsible for interpreting the user's input and calling the respective command. 
+
+### Command
+
+The different `Command` classes are responsible for receiving the user's input from the `Parser` class and executing the corresponding commands. 
+The figure belows shows the class diagram of the command class: <br/>
+<img src="https://raw.githubusercontent.com/AY2021S1-CS2113T-T12-2/tp/master/images/command.JPG" alt="" width="200"/><br/>
```

Comment:
![image](https://user-images.githubusercontent.com/18654251/97659384-d5946700-1aa9-11eb-85ce-9d3a31b0d77c.png)

Perhaps you can expand this class diagram by showing how each Command class will interact or inherit from the abstract Command class.